libres/lib/analysis/stepwise.cpp

```diff
@@ -8,370 +8,356 @@
+                    for (irow = validation_start; irow <= validation_end;
+                         irow++) {
+                        matrix_copy_row(x_vector, stepwise->X0, 0,
+                                        randperms[irow]);
+                        //matrix_copy_row( e_vector , stepwise->E0 , 0 , randperms[irow]);
+                        {
+                            double true_value =
+                                matrix_iget(stepwise->Y0, randperms[irow], 0);
+                            double estimated_value =
+                                stepwise_eval__(stepwise, x_vector);
+                            prediction_error += (true_value - estimated_value) *
+                                                (true_value - estimated_value);
+                            //double e_estimated_value = stepwise_eval__( stepwise , e_vector );
+                            //prediction_error += e_estimated_value*e_estimated_value;
+                        }
+                    }
+                    matrix_free(x_vector);
+                }
             }
-
-          }
-          matrix_free( x_vector );
         }
-      }
-    }
 
-    free( randperms );
-    bool_vector_free( active_rows );
-  }
+        free(randperms);
+        bool_vector_free(active_rows);
+    }
 
-  /*inactivate the test_var-variable after completion*/
-  bool_vector_iset( stepwise->active_set , test_var , false );
-  return prediction_error;
+    /*inactivate the test_var-variable after completion*/
+    bool_vector_iset(stepwise->active_set, test_var, false);
+    return prediction_error;
 }
 
+void stepwise_estimate(stepwise_type *stepwise, double deltaR2_limit,
+                       int CV_blocks) {
+    int nvar = matrix_get_columns(stepwise->X0);
+    int nsample = matrix_get_rows(stepwise->X0);
+    double currentR2 = -1;
+    bool_vector_type *active_rows = bool_vector_alloc(nsample, true);
 
-void stepwise_estimate( stepwise_type * stepwise , double deltaR2_limit , int CV_blocks) {
-  int nvar          = matrix_get_columns( stepwise->X0 );
-  int nsample       = matrix_get_rows( stepwise->X0 );
-  double currentR2 = -1;
-  bool_vector_type * active_rows = bool_vector_alloc( nsample , true );
-
-
-  /*Reset beta*/
-  for (int i = 0; i < nvar; i++) {
-    matrix_iset(stepwise->beta, i , 0 , 0.0);
-  }
-
-
+    /*Reset beta*/
+    for (int i = 0; i < nvar; i++) {
+        matrix_iset(stepwise->beta, i, 0, 0.0);
+    }
 
-  bool_vector_set_all( stepwise->active_set , false );
+    bool_vector_set_all(stepwise->active_set, false);
 
-  double MSE_min = 10000000;
-  double Prev_MSE_min = MSE_min;
-  double minR2    = -1;
+    double MSE_min = 10000000;
+    double Prev_MSE_min = MSE_min;
+    double minR2 = -1;
 
-  while (true) {
-    int    best_var = 0;
-    Prev_MSE_min = MSE_min;
+    while (true) {
+        int best_var = 0;
+        Prev_MSE_min = MSE_min;
 
-    /*
+        /*
```

Comment:
The indentation of this comment looks a bit funky since indentation of the code has changed.